CUDADataFormats/TrackingRecHit/interface/TrackingRecHitSoADevice.h

```diff
@@ -0,0 +1,97 @@
+#ifndef CUDADataFormats_RecHits_TrackingRecHitsDevice_h
+#define CUDADataFormats_RecHits_TrackingRecHitsDevice_h
+
+#include <cstdint>
+
+#include "CUDADataFormats/TrackingRecHit/interface/TrackingRecHitsUtilities.h"
+#include "CUDADataFormats/Common/interface/PortableDeviceCollection.h"
+#include "HeterogeneousCore/CUDAUtilities/interface/host_unique_ptr.h"
+#include "HeterogeneousCore/CUDAUtilities/interface/cudaCheck.h"
+
+template <typename TrackerTraits>
+class TrackingRecHitSoADevice : public cms::cuda::PortableDeviceCollection<TrackingRecHitLayout<TrackerTraits>> {
+public:
+  using hitSoA = TrackingRecHitSoA<TrackerTraits>;
+  //Need to decorate the class with the inherited portable accessors being now a template
+  using cms::cuda::PortableDeviceCollection<TrackingRecHitLayout<TrackerTraits>>::view;
+  using cms::cuda::PortableDeviceCollection<TrackingRecHitLayout<TrackerTraits>>::const_view;
+  using cms::cuda::PortableDeviceCollection<TrackingRecHitLayout<TrackerTraits>>::buffer;
+  using cms::cuda::PortableDeviceCollection<TrackingRecHitLayout<TrackerTraits>>::bufferSize;
+
+  TrackingRecHitSoADevice() = default;  // cms::cuda::Product needs this
+
+  using AverageGeometry = typename hitSoA::AverageGeometry;
+  using ParamsOnGPU = typename hitSoA::ParamsOnGPU;
+  using PhiBinnerStorageType = typename hitSoA::PhiBinnerStorageType;
+  using PhiBinner = typename hitSoA::PhiBinner;
+  // Constructor which specifies the SoA size
+  explicit TrackingRecHitSoADevice(uint32_t nHits,
+                                   int32_t offsetBPIX2,
+                                   ParamsOnGPU const* cpeParams,
+                                   uint32_t const* hitsModuleStart,
+                                   cudaStream_t stream)
+      : cms::cuda::PortableDeviceCollection<TrackingRecHitLayout<TrackerTraits>>(nHits, stream),
+        nHits_(nHits),
+        cpeParams_(cpeParams),
+        hitsModuleStart_(hitsModuleStart),
+        offsetBPIX2_(offsetBPIX2) {
+    cudaCheck(cudaGetLastError());
+    cudaCheck(cudaDeviceSynchronize());
```

Comment:
is this really needed ?

if it' needed, could it be replaced with `cudaCheck(cudaStreamSynchronize(stream))` ?